### PR TITLE
Bump to v4.12.0

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**

### CODE REVIEW

File: SphereEversion/ToMathlib/Analysis/Calculus.lean [hunk 1]
```diff
@@ -32,7 +32,8 @@ theorem ContDiffAt.comp₂ {g : E₁ × E₂ → G} {f₁ : F → E₁} {f₂ :
 
 theorem ContDiffAt.clm_comp {g : E' → F →L[𝕜] G} {f : E' → E →L[𝕜] F} {n : ℕ∞} {x : E'}
     (hg : ContDiffAt 𝕜 n g x) (hf : ContDiffAt 𝕜 n f x) : ContDiffAt 𝕜 n (fun x ↦ g x ∘L f x) x :=
-  isBoundedBilinearMap_comp.contDiff.contDiffAt.comp₂ hg hf
+  isBoundedBilinearMap_comp.contDiff.contDiffAt.comp₂
+    (g := fun p => ContinuousLinearMap.comp p.1 p.2) hg hf
```
Comment (PatrickMassot) [inline SphereEversion/ToMathlib/Analysis/Calculus.lean:36]: Do you understand what happened here?

File: SphereEversion/Global/OneJetBundle.lean [hunk 2]
```diff
@@ -316,7 +316,8 @@ theorem oneJetBundle_chart_target (x₀ : J¹MM') :
       (Prod.map (chartAt H m).symm (chartAt H' m').symm) ∘ Prod.fst := by
     ext x <;> rfl
   rw [this, preimage_comp, preimage_prod_map_prod]
-  mono
+  simp only [Prod.range_fst, subset_univ, preimage_subset_preimage_iff]
+  gcongr
```
[the remark this answers is not in the document]
Comment (Ruben-VandeVelde) [reply SphereEversion/Global/OneJetBundle.lean:318]: It turns out that importing `Mathlib.Tactic.Monotonicity.Lemmas` makes the old proof work again. I'm happy to go either way here.

File: SphereEversion/ToMathlib/Analysis/Calculus.lean (hunk 1 above)
Comment (Ruben-VandeVelde) [reply SphereEversion/ToMathlib/Analysis/Calculus.lean:36]: I don't. My best guess is that it's one of those hard unification problems and that we just got lucky before.

File: SphereEversion/Global/OneJetBundle.lean (hunk 2 above)
[the remark this answers is not in the document]
Comment (PatrickMassot) [reply SphereEversion/Global/OneJetBundle.lean:318]: I would like to get back the old proof, yes.

[the remark this answers is not in the document]
Comment (grunweg) [reply SphereEversion/Global/OneJetBundle.lean]: I restored the old proof.

**Pull request merged**